mostlyai/engine/_tabular/interface.py

```diff
@@ -632,126 +631,87 @@ def predict_proba(
+               These features are used as seed_data to condition the probability generation.
+            target: Name of the target column(s) to predict. Can be:
+                - str: Single target column name
+                - list[str]: Multiple target columns for joint probabilities (supports arbitrary number of targets)
+                - None: Uses the target column from fit()
             ctx_data: Context data for generation. If None, uses the context data from training.
-            n_draws: Number of draws to generate for each sample during prediction. Defaults to 1.
-            **kwargs: Additional arguments passed to sample() method.
+            rare_category_replacement_method: How to handle rare categories:
+                - "constant" (default): Suppress rare tokens in probability outputs
+                - "sample": Always suppress rare tokens for categorical columns
+            **kwargs: Additional generation parameters (device, etc.).
 
         Returns:
-            pd.DataFrame with shape (n_samples, n_classes) where columns are named by class labels.
-            Column names are derived from encoding stats:
-            - Binned: Special tokens (e.g., "<<NULL>>"), min value, bin ranges (e.g., "<10000", ">=50000"), and max value (e.g., "0", ..., "99999")
-            - Categorical: All category names including special tokens (e.g., "_RARE_", "<<NULL>>", "male", "female")
-            - Discrete: All numeric values including special tokens (e.g., "_RARE_", "<<NULL>>", "1", "2", "3")
-            Each row contains probability distribution that sums to 1.0.
+            Single target: DataFrame with shape (n_samples, n_categories) containing probability distributions.
+                Column names are derived from encoding stats:
+                - Binned: "<{value}" or ">={value}" (e.g., "<10000", ">=10000")
+                - Categorical: category names (e.g., "male", "female")
+                - Discrete: discrete values (e.g., "0", "1", "2")
+                Each row contains a full probability distribution that sums to 1.0.
+
+            Multiple targets: MultiIndex DataFrame with joint probabilities.
+                Columns: MultiIndex.from_product of all category combinations.
+                Example for target=["gender", "income"]:
+                    gender  income
+                    male    <10000     0.15
+                            >=10000    0.35
+                    female  <10000     0.25
+                            >=10000    0.25
 
         Raises:
-            ValueError: If target column has unsupported encoding type.
+            ValueError: If model is not fitted, target is not specified, or target has unsupported
+                       encoding type for probability output.
+
+        Notes:
+            Performance: Joint probability complexity is O(card1 × card2 × ... × cardN),
+            exponential in the number of targets.
```

Comment:
This may be easily overlooked. I'd consider at least a clear warning, if not some log inside the function, if we expect a very lengthy (and mem-consuming?) computation.